forge/housekeeper/tasks/certifiedNodes.js

```diff
@@ -0,0 +1,119 @@
+                        if (mod.startsWith(SCOPE_CERTIFIED) || mod.startsWith(SCOPE_NODES)) {
+                            if (payload.certifiedNodes[mod]) {
+                                payload.certifiedNodes[mod]++
+                            } else {
+                                payload.certifiedNodes[mod] = 1
+                            }
+                        }
+                    }
+                }
+            }
+
+            // all running devices seen in the last 24hrs
+            const now = Date.now()
+            const runningDevices = await app.db.models.Device.findAll({
+                where: {
+                    state: 'running',
+                    lastSeenAt: { [Op.gte]: new Date(now - 1000 * 60 * 60 * 24) }
+                }
+            })
+            for (const dev of runningDevices) {
+                const activeSnapshot = dev.activeSnapshotId
+                if (activeSnapshot !== undefined) {
+                    if (snapshots[activeSnapshot]) {
+                        const cache = snapshots[activeSnapshot]
+                        for (const mod of cache) {
+                            if (payload.certifiedNodes[mod]) {
+                                payload.certifiedNodes[mod]++
+                            } else {
+                                payload.certifiedNodes[mod] = 1
+                            }
+                        }
+                    } else {
+                        const snapshot = await app.db.models.ProjectSnapshot.byId(activeSnapshot)
+                        if (snapshot.settings.modules) {
+                            const cache = []
+                            for (const mod of Object.keys(snapshot.settings.modules)) {
+                                if (mod.startsWith(SCOPE_CERTIFIED) || mod.startsWith(SCOPE_NODES)) {
+                                    cache.push(mod)
+                                    if (payload.certifiedNodes[mod]) {
+                                        payload.certifiedNodes[mod]++
+                                    } else {
+                                        payload.certifiedNodes[mod] = 1
+                                    }
+                                }
+                            }
+                            snapshots[activeSnapshot] = cache
+                        }
+                    }
+                }
+            }
+        } catch (err) {
+            app.log.error('Failed to gather Certified Nodes usage', err)
+        }
+
+        // decide where to send response
+        try {
+            await axios.post(CERTIFIED_NODES_ENDPOINT, payload)
+        } catch (err) {
+            app.log.error('Failed to upload Certified Nodes usage', err)
+        }
+    } else {
+        // console.log('NOT ACTIVE')
+        // console.log(catalogueString, npmRegURLString, token)
+    }
+}
+
+module.exports = {
+    name: 'certifiedNodes',
+    startup: 15000,
```

Comment:
```suggestion
```
Given the workload involved, we don't need to be reporting each time the app restarts